### PR TITLE
Delta Protocol adherence

### PR DESCRIPTION
**Pull request by Jonathan-Rosenberg**
https://books.japila.pl/delta-lake-internals/CommitInfo/
https://books.japila.pl/delta-lake-internals/Operation/#parameters

Change the `operationParameters` type to map[string]any

### CODE REVIEW

**Review by guy-har — APPROVED** [review]

File: parquet_marshaller.go [hunk 1]
```diff
@@ -322,7 +323,15 @@ func parquetMarshalCommitInfo(add *action.CommitInfo, obj interfaces.MarshalObje
 		obj.AddField("userName").SetByteArray([]byte(*add.UserName))
 	}
 	obj.AddField("operation").SetByteArray([]byte(add.Operation))
-	parquet.MarshalMap(obj, "operationParameters", add.OperationParameters)
+	ops := make(map[string]string, len(add.OperationParameters))
+	for k, v := range add.OperationParameters {
+		if vs, ok := v.(string); !ok {
+			ops[k] = fmt.Sprintf("%v", v)
```
Comment (nopcoder) [inline parquet_marshaller.go:329]: fmt.Sprint

File: parquet_marshaller.go [hunk 2]
```diff
@@ -360,6 +369,13 @@ func parquetMarshalCommitInfo(add *action.CommitInfo, obj interfaces.MarshalObje
 }
 
 func parquetUnmarshalCommitInfo(add *action.CommitInfo, obj interfaces.UnmarshalObject) error {
+	// commitInfo isn't included in a checkpoint parquet file:
```
Comment (nopcoder) [inline parquet_marshaller.go:372]: move down to the where we use `commitInfo`

File: trnx.go [hunk 3]
```diff
@@ -190,14 +190,18 @@ func (trx *optimisticTransactionImp) Commit(actionsIter iter.Iter[action.Action]
 	}
 	isBlindAppend := !dependsOnFiles && onlyAddFiles
 
+	opAny := make(map[string]any, len(op.Parameters))
+	for k, v := range op.Parameters {
+		opAny[k] = v
+	}
```
Comment (nopcoder) [inline trnx.go:196]: please check why we need to convert from map[string]string as operation should hold map[string]any

**Review by nopcoder — CHANGES_REQUESTED** [review]

File: parquet_marshaller.go [hunk 4]
```diff
@@ -368,14 +364,14 @@ func parquetMarshalCommitInfo(add *action.CommitInfo, obj interfaces.MarshalObje
 	return nil
 }
 
+/**
+ * commitInfo isn't included in a checkpoint parquet file:
```
Comment (nopcoder) [inline parquet_marshaller.go:368]: - use `//` for function documentation.
- start with function name - parquetUnmarshalCommitInfo

**Review by nopcoder — CHANGES_REQUESTED** [review]

**Pull request merged**